src/main/java/abdulhakov/eldar/CustomList.java

```diff
@@ -0,0 +1,82 @@
+package abdulhakov.eldar;
+
+import java.util.AbstractList;
+
+public class CustomList<T> extends AbstractList<T> {
+
+    private Object[] array = new Object[0];
+    private int size = 0;
+
+    @Override
+    public boolean add(T value) {
+        Object[] newArray = new Object[size + 1];
```

Comment:
deeply inefficient, meaning: 
if you have 1_000_000 elements in the array, once you add the element #1_000_001, the code will copy 1M elements just to add one. And will do this every time. 
In this case, time complexity of every add() operation is O(N), which is slow

please address